privileges/privileges_darwin.go

```diff
@@ -0,0 +1,12 @@
+//go:build darwin
```

Comment:
The checks need to be more generic as now they are not contextualized anymore to naabu only. This privilege check is valid for both linux and darwin